dashboard/app/views/levels/_level_group.haml

```diff
@@ -27,14 +27,22 @@
 
     -# Show any text from an external level.
 
+    -# Fall back to retrieving the the last_attempt from the parsed levelgroup last_attempt
+    - sublevel_last_attempt = parsed[level.id.to_s].try(:[], 'result')
+    - if @user && current_user && @user != current_user
+      -# load other user's solution for teachers viewing their students' solution
+      - sublevel_last_attempt = @user.last_attempt(level).try(:level_source).try(:data)
+    - elsif current_user
+      -# load user's previous attempt at this puzzle.
+      - sublevel_last_attempt = current_user.last_attempt(level).try(:level_source).try(:data)
```

Comment:
Given that we have a fallback result coming from parsed, are we sure we're getting coverage on this code?